### ISSUE TITLE
The `declaration-block-semicolon-*` rules report a declaration without a semicolon where a comment closes the block, and the `-before` fixes grow the file on every run

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#208]
Found while fixing [#203](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/203), and unrelated to it: the flag plays no part, and `main` behaves the same before and after that fix.

### What minimal example or steps are needed to reproduce the bug?

```css
a { color: pink /* c */ }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/declaration-block-semicolon-newline-before": "always"
  }
}
```

The `declaration-block-semicolon-space-before` rule does the same under its `always` options.

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

Nothing reported and nothing written: the declaration has no semicolon, so there is no whitespace in front of one to measure, and the last declaration of a block is passed over as soon as nothing stands behind it.

### What actually happened?

Without `--fix`, `1:15 Expected newline before ";"` is reported for a semicolon the file does not hold. With it, a break is written in front of the comment, and another one on every further run, since the break lands in the comment's own whitespace and the declaration is measured the same way again:

```
run 1   a { color: pink\n /* c */ }
run 2   a { color: pink\n\n /* c */ }
run 3   a { color: pink\n\n\n /* c */ }
```

The `space-before` sibling grows a space at a time the same way, `a { color: pink  /* c */ }`, then three spaces, then four; the `always-multi-line` and `always-single-line` options and an inline `//` comment under `postcss-scss` behave alike. The `never*` options of both rules are not affected.

### What is the cause?

Both rules skip the declaration that closes a block without a semicolon with `!parentRule.raws.semicolon && parentRule.last === decl`, and a comment standing between that declaration and the closing brace is a node of its own, the last one, so the declaration is no longer the block's last and the guard lets it through. The last declaration of the block is what wants to be found — the last node that is a declaration, not the last node of any kind.

### Update

The guard is the same line in all four `declaration-block-semicolon-*` rules, and `declaration-block-semicolon-space-after` misreads the same file: under `always` the comment of `a { color: pink\n  /* c */\n}` is pulled up to the declaration, and under `never` the space of `a { color: pink /* c */ }` is closed up — a fix for a semicolon that is not there, though a bounded one. `declaration-block-semicolon-newline-after` carries the guard too, but steps past the comments on its own and is not affected. The fix answers the four at once, so this issue covers the `-space-after` rule as well.
